source/common/ssl/context_manager_impl.cc

```diff
@@ -20,18 +20,19 @@ void ContextManagerImpl::releaseContext(Context* context) {
   contexts_.remove(context);
 }
 
-ClientContextPtr ContextManagerImpl::createSslClientContext(Stats::Scope& scope,
-                                                            const ClientContextConfig& config) {
-  ClientContextPtr context(new ClientContextImpl(*this, scope, config));
+ClientContextSharedPtr
+ContextManagerImpl::createSslClientContext(Stats::Scope& scope, const ClientContextConfig& config) {
+  ClientContextSharedPtr context(new ClientContextImpl(*this, scope, config));
   std::unique_lock<std::shared_timed_mutex> lock(contexts_lock_);
   contexts_.emplace_back(context.get());
```

Comment:
since now contexts are shared_ptr, shouldn't `contexts_` be a list of `weak_ptr`?

Reply:
There is not easy way to remove the item in the list after its weak_ptr ref_count is 0.  Then there will be weak_ptr leak.

Reply:
Done. releaseContext is removed.